src/App.tsx

```diff
@@ -15,33 +16,131 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPABET) {
+    visibleGoods.sort((g1, g2) => {
+      return g1.localeCompare(g2);
+    });
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((g1, g2) => {
+      return g1.length - g2.length;
+    });
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const start = () => setIsStarted(true);
+  const reverse = () => setIsReversed(!isReversed);
+  const alphabet = () => setSortType(SortType.ALPABET);
```

Comment:
functions should be named as actions they perform 😉 
```suggestion
  const sortByAlphabet = () => setSortType(SortType.ALPABET);
```

Reply:
variable `alphabet` looks like you just keep a string `'abcd...xyz'`